### PR TITLE
Authentication/Account Servlets Testing

### PR DESCRIPTION
**Pull request by azhara-a**
**Write AuthenticationServletTest and AccountServletTest with mockito and AppEngine Testing library**
- Change User class name to UserInfo as it was clashing with com.google.appengine.api.User.
- Add additional constructor to AuthenticationServlet with a parameter UserService to enable mocking it during testing.
- Add parameter UserService to getUserEntity() to enable mocking it.

### CODE REVIEW

**Review by mam249 — COMMENTED** [review]
LGTM

File: backend/src/main/java/com/google/sps/servlets/AuthenticationServlet.java [hunk 1]
```diff
@@ -29,12 +29,22 @@
...
 public abstract class AuthenticationServlet extends HttpServlet {
-  private static final int AUTHENTICATION_ERROR_CODE = 401;
+  public static final int AUTHENTICATION_ERROR_CODE = 401;
+  protected static UserService mUserService;
+
+  public AuthenticationServlet() {
+    mUserService = UserServiceFactory.getUserService();
+  }
+
+  // For testing, UserService is mocked
```
Comment (mugdhalakhani) [inline backend/src/main/java/com/google/sps/servlets/AuthenticationServlet.java:39]: can you instead make a function called setUserServiceForTesting() and annotate it with @VisibleForTesting?

File: backend/src/test/java/com/google/sps/servlets/AccountServletTest.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  @Test
+  public void getReturnsUserDetailsWithNoDietsAndAllergies() throws Exception {
+    Entity user = TestUtils.getUserEntityWithName();
+    UserInfo expectedUser =
+        new UserInfo(TestUtils.NAME, Collections.emptyList(), Collections.emptyList());
+    getReturnsUserDetails(user, expectedUser);
+  }
+
+  @Test
+  public void postCreatesNewUserAndAddsToDatastore() throws Exception {
```
Comment (mugdhalakhani) [inline backend/src/test/java/com/google/sps/servlets/AccountServletTest.java:127]: perhaps in a followup, consider adding malformed post requests to verify expected behavior.

**Review by mugdhalakhani — APPROVED** [review]
Thanks!

File: backend/src/test/java/com/google/sps/servlets/AuthenticationServletTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    verify(response).setStatus(AuthenticationServlet.AUTHENTICATION_ERROR_CODE);
+  }
+
+  @Test
+  public void loggedInGetIsExecuted() throws Exception {
+    when(userService.isUserLoggedIn()).thenReturn(true);
+    String result =
+        TestUtils.getResultFromAuthenticatedGetRequest(
+            getTestAuthenticationServlet(), request, response);
+    assertEquals(result, "in get");
```
Comment (lminchillo) [inline backend/src/test/java/com/google/sps/servlets/AuthenticationServletTest.java:85]: nit: make "in get" and "in post" constants in this class

**Review by lminchillo — APPROVED** [review]
LGTM, feel free to resolve comments in a followup PR

**Pull request merged**